### DISCUSSION TITLE
Support - Emails not sending with latest version

### DISCUSSION THREAD

**Discussion by NikRimington** [nathanwoulfe/Plumber-2#89] [Q&A]
Hi Nathan,

We've updated a site to the latest version of Plumbr (v1.6.2.574) on Umbraco 8.14.0 and our client has reported that emails are not longer being sent. We cannot see any errors relating to emailing in the Umbraco logs and there is no evidence of the emails being sent through our SMTP server.

What settings need checking with regards to email sending to ensure they get sent?

Thanks

Nik

**Comment by nathanwoulfe** [comment]
Hey @NikRimington, that's a bit annoying...

There were substantial changes in the 1.6 release to introduce editable HTML email templates, so quite possible there's a bug snuck in somewhere. First things I would check is that all the settings on the Notifications tab (Workflow => Settings => Notifications app) are correct - can't go too far wrong here though. Also check that the email views exist on disk (/views/partials/workflowemails) and that the corresponding variant views exist (if required), they should have been installed by the package upgrade, could be missing from your deployed site?

I had hoped to have 1.6.3 out by now, but got sidetracked with v9 migration, so now it looks like 1.6.3 will be 1.7 (or more likely 8.0 so that the versioning can align with Umbraco's majors).

Let me know if the above helps...

N

**Reply by NikRimington** [reply]
Hi @nathanwoulfe ,

Thanks for coming back to me. I can see that the email templates have all been deployed.
![image](https://user-images.githubusercontent.com/13313745/124719490-1bd67b80-deff-11eb-8acc-506998d76a08.png)

The settings in the Notifications app look like this:
![image](https://user-images.githubusercontent.com/13313745/124719687-4f190a80-deff-11eb-9f25-94a24fbc1642.png)

The site is single language, so I think everything is being treated as "invariant" looking at the History tab.

What namespace would I use if I wanted to explicily turn on debug logging for Plumbr code? Would it just be "Plumbr" ?

Thanks,

Nik

**Reply by nathanwoulfe** [reply]
Wondering if I've done a dodgy with the notification delay property... Maybe try setting that to a large value (100 or so) so that the reminders are never triggered, but the check for sending notifications would pass (I'm guessing this might be the reason, will check code when I can)

Plumber.Core would be enough for logging, might get noisy with Plumber.Web as well. I'll double check for a more specific namespace too. 

Will find my 1.6.2 branch and see what I can replicate

**Reply by nathanwoulfe** [reply]
Reminders delay should be fine left as 0 - logic around that is ok. I have a stack of local emails in my drop folder, but looking through those they all appear to be reminders, not workflow notifications. It's a bit strange that all the tests would pass, and my debug API works, but looks like they're not sending...

Will start digging...

**Reply by nathanwoulfe** [reply]
And one more for tonight - I've found a couple of buggy bits which will need fixing. I'd rather bundle these into the next release as opposed to patching, since there are a stack of related improvements in the coming minor.

I won't get to it tonight, but am aiming for at least a beta for the next minor this week. Would your client be ok with a few days wait?

**Reply by NikRimington** [reply]
They will have to :-D

**Reply by nathanwoulfe** [reply]
@NikRimington by the time you read this, 1.6.3-beta--000643 will be on nuget. Pretty sure it's working correctly now - I've added all email-to options for all actions, and pumped through a few workflows, emails all hit my mail folder.

**Reply by NikRimington** [reply]
I'll try and get this installed today/tomorrow :-)

**Reply by NikRimington** [reply]
Hey @nathanwoulfe I updated to the beta of 1.6.3, and it got deployed out. However it's not being reported to the sending notifications for approvals for pages that don't have anything pending approval. 

What could cause that to happen? I've had a look in the DB tables and can't see a reason for this notification being sent. One of the example emails the client provided us was for a node that was last processed 7 days ago and that workflow was completed half an hour after it was triggered.

**Reply by nathanwoulfe** [reply]
Strange... Was the email a notification re the pending task, or a reminder for an overdue task?

Will need to have a look, but there's no good reason for old tasks to have notifications sent, but it could be an issue with the scheduled task for sending reminders.

**Reply by NikRimington** [reply]
We've not customised any of the email templates/messages. The Subject of the email was "Workflow: Approval request"

So I have no idea what type of email it was exactly, but the content item it linked to didn't have any pending tasks or overdue ones that I could see.

I've also noticed the link in the email contains "http://127.0.0.1" instead of the site domain? Not sure if that is a configuration issue that I'm missed setting something? (sorry I suspect that's unrelated to this discussion).

**Reply by nathanwoulfe** [reply]
Has it only sent the single erroneous email? It's particularly odd since the logic for triggering the notifications only exists in the workflow process code, it isn't accessible without a task being processed...

As for the 127.0.0.1, I think I may have merged down a v9 change - to fix it, setting the site URL and edit site URL on the notifications tab in workflow settings. Previously these values were extracted from HttpContext, but it started getting messy dealing with 8 and 9 and background tasks, where there is no context, so figured it would be easier to make them required settings. 

This should have been in the next major, I think. I'll add to the release notes.
